### PR TITLE
[Bug] Fix the flink on k8s application build failed and savepointPath saved failed bug when use h2 as database

### PR DESCRIPTION
**Pull request by youyangkou**
#### fix the flink on k8s application build failed and savepointPath save failed bug when use h2 as database

Problem Summary:
1. because the initSetting has @PostConstruct in SettingServiceImpl class ,the variable settings size is 0 when  streampark console start.
so when update setting the code ```settings.get(setting.getSettingKey()).setSettingValue(value)``` will case Nullpoint Exception and the variable settings'size will  be 0 Always.

2. the table t_flink_savepoint in H2 missing a column 'chk_id'。
### What is changed and how it works?

Proposal: [xxx](url) 

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/streampark/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java [hunk 1]
```diff
@@ -81,6 +74,9 @@ public boolean update(Setting setting) {
             if (setting.getSettingKey().equals(CommonConfig.MAVEN_AUTH_PASSWORD().key())) {
                 InternalConfigHolder.set(CommonConfig.MAVEN_AUTH_PASSWORD(), value);
             }
+            if (settings == null || settings.size() == 0) {
```
Comment (lvshaokang) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java]: Is it necessary to init settings again?

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java [hunk 2]
```diff
@@ -81,6 +74,9 @@ public boolean update(Setting setting) {
...
+                initSetting();
+            }
             settings.get(setting.getSettingKey()).setSettingValue(value);
```
Comment (lvshaokang) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java]: if not, suggest use optional to resolve npe. btw, settings is not null, refer to 51 line.
```suggestion
            Optional<Setting> optional = Optional.ofNullable(settings.get(setting.getSettingKey()));
            if (optional.isPresent()) {
                optional.get().setSettingValue(value);
            }
```

File: streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql [hunk 3]
```diff
@@ -197,6 +197,7 @@ create table if not exists `t_flink_project` (
 create table if not exists `t_flink_savepoint` (
   `id` bigint generated by default as identity not null,
   `app_id` bigint not null,
+  `chk_id` bigint default null,
```
Comment (lvshaokang) [inline streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql:200]: Plz check pgsql and mysql dialect whether lack field or without default value.

**Review by lvshaokang — CHANGES_REQUESTED** [review]

File: streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql (hunk 3 above)
Comment (youyangkou) [reply streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql:200]: > Plz check pgsql and mysql dialect whether lack field or without default value.

mysql and pgsql have chk_id，this is data-h2.sql bug.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java (hunk 2 above)
Comment (youyangkou) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java]: @lvshaokang Thanks for your review, addressed.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java (hunk 1 above)
Comment (youyangkou) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java]: > Is it necessary to init settings again?

It is necessary, the settings are empty when the data has not been loaded in the H2 database

Comment (youyangkou) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java]: > Is it necessary to init settings again?

You can test it in H2 database schema

**Comment by youyangkou** [comment]
@wolfboys PTAL

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java [hunk 4]
```diff
@@ -61,27 +62,27 @@ public void initSetting() {
...
+            if (CommonConfig.MAVEN_AUTH_USER().key().equals(settingKey)) {
                 InternalConfigHolder.set(CommonConfig.MAVEN_AUTH_USER(), value);
             }
-            if (setting.getSettingKey().equals(CommonConfig.MAVEN_AUTH_PASSWORD().key())) {
+            if (CommonConfig.MAVEN_AUTH_PASSWORD().key().equals(settingKey)) {
                 InternalConfigHolder.set(CommonConfig.MAVEN_AUTH_PASSWORD(), value);
             }
-            settings.get(setting.getSettingKey()).setSettingValue(value);
+            if (settings == null || settings.size() == 0) {
+                initSetting();
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java]: This logic is not as expected. H2, MySQL, and pgsql should be consistent. Data in data-h2.sql will be inserted into `t_ setting` table. These are completed in the system initialization phase, initSetting will only run once, So `initSetting` method should not be called again

Comment (youyangkou) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java]: Thanks for your review, addressed.

**Review by wolfboys — APPROVED** [review]
LGTM

**Review by lvshaokang — APPROVED** [review]
@youyangkou LGTM, Thanks!

**Pull request merged**